proposals/0387-cross-compilation-destinations.md

```diff
@@ -167,75 +181,165 @@ that, a separate `toolset.json` file is introduced:
 `swift build --toolset toolset1.json --toolset toolset2.json` will build with `/custom/swiftc` and no extra flags, as
 specified in `toolset2.json`, but `/usr/bin/clang -pedantic` from `toolset1.json` will still be used.
 
 Tools not specified in any of the supplied toolset files will be looked up in existing implied search paths that are
-used without any presence of toolset files, even when `toolsetRootPath` is present in any toolset file. We'd like
-toolsets to be explicit enough in this regard: if any tool would like to participate in toolsets path lookups, it must
-have its path provided, either relative or absolute, in any of the provided toolset files.
+used without toolsets, even when `toolsetRootPath` is present. We'd like toolsets to be explicit in this regard: if a
+tool would like to participate in toolset path lookups, it must provide either a relative or an absolute path in a
+toolset.
+
+Tools that don't have `path` property but have `extraCLIOptions` present will append options from that property to a
+tool with the same name specified in a preceding toolset file. If no other toolset files were provided, these options
+will be appended to the default tool invocation. For example `pedanticCCompiler.json` that looks like this
+
+```json5
+{
+  "schemaVersion": "1.0",
+  "cCompiler": {
+    "extraCLIOptions": ["-pedantic"]
+  }
+}
+```
+
+in `swift build --toolset pedanticCCompiler` will pass `-pedantic` to the C compiler located at a default path.
+
+When cross-compiling, paths in `toolset.json` files supplied in destination artifact bundles should be self-contained:
+no absolute paths and no escaping symlinks are allowed. Users are still able to provide their own `toolset.json` files
+outside of artifact bundles to specify additional developer tools for which no relative "non-escaping" path can be
+provided within the bundle.
+
+### `destination.json` Files
+
+Note the presence of `destination.json` files in each `<destination artifact>` subdirectory. These files should contain
+a JSON dictionary with an evolved version of the schema of [existing `destination.json` files that SwiftPM already
+supports](https://github.com/apple/swift-package-manager/pull/1098) and `destination.json` files presented in the pitch
+version of this proposal, hence `"schemaVersion": "3.0"`:
 
-When cross-compiling, paths in `toolset.json` files supplied in destination artifact bundles should be self-contained
-with same rules applied as in `destination.json`: no absolute paths and no escaping symlinks are allowed. Users are
-still able to provide their own `toolset.json` files outside of artifact bundles to specify additional developer tools
-for which no relative path can be provided within the bundle. 
+```json5
+{
+  "schemaVersion": "3.0",
+  "runTimeTriples": [
+    "<triple1>": {
+      "swiftResourcesPaths": ["<array of paths relative to `destination.json` containing Swift resources>"],
```

Comment:
The Swift compiler doesn't use multiple resource directories per build, only using the last one specified on the CLI, so this should be singular, now that it's only for a single runtime triple.